core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -137,6 +137,12 @@ private[spark] class TaskSetManager(
   // Task index, start and finish time for each task attempt (indexed by task ID)
   val taskInfos = new HashMap[Long, TaskInfo]
 
+  val successfulTasksSet = new scala.collection.mutable.TreeSet[Long] {
```

Comment:
this can't be a set, since multiple tasks might be running for the same amount of time.  you could change it to a set of `(time, count)` pairs, which would make the update logic a bit more complicated, or just keep a list of all runtimes.

also the name should indicate that its the durations, eg. `successfulTaskDurations`